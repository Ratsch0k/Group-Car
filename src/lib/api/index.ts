export * from './getRandomProfilePic';
export * from './group';
export * from './request.d';
export * from './userType.d';
export * from './auth';
export * from './user';
