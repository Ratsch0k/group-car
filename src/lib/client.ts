import axios, {AxiosResponse} from 'axios';

const csrfPromise = (() => {
  return axios.head('/auth').then((res: AxiosResponse) => {
    const csrf = res.headers['xsrf-token'];

    axios.defaults.headers.common = {
      'xsrf-token': csrf,
    };
  });
})();

const methodsToProxy = ['post', 'put', 'get', 'head', 'delete', 'options'];

methodsToProxy.forEach((method) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const origMethod = (axios as any)[method];

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (axios as any)[method] = (...args: any[]) => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return csrfPromise.then(() => origMethod(...args));
  };
});

export default axios;
