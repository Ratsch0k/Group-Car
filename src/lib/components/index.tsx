export * from './Authentication';
export * from './CloseableDialogTitle';
export * from './Drawer';
export * from './DrawerCard';
export * from './HeaderBar';
export * from './Input';
export * from './UserAvatar';
export * from './UserOverview';
export * from './legal';
export * from './AutoFullscreenDialog';
export * from './AxiosErrorHandler';
