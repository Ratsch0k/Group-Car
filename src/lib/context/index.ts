export * from './axiosContext';
export * from './apiContext';
export * from './invitesContext';
export * from './snackbarContext';
export * from './mapContext';
