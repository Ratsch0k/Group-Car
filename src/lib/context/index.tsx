export * from './authContext';
export * from './axiosContext';
