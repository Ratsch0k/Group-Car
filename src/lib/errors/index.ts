export * from './NotDefinedError';
export * from './CouldNotDriveCarError';
export * from './CouldNotParkCarError';
export * from './CarAlreadyExistsError';
export * from './NotLoggedInError';
export * from './NoInviteForGroupError';
