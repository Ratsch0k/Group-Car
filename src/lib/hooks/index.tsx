export {default as useModalRouter} from './useModalRouter';
export * from './useAxios';
