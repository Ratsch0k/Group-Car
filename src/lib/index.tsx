export * from './components';
export * from './ModalRouter';
export * from './theme';
export * from './requests';
export * from './Routes';
export * from './context';
export * from './hooks';
