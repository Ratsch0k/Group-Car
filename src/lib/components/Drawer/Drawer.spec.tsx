import React from 'react';
import {render} from '@testing-library/react';
import {ThemeProvider} from '@material-ui/core';
import {theme, Drawer} from 'lib';

it('renders and matches snapshot with open and ' +
    'not permanent without crashing', () => {
  const {baseElement} = render(
      <ThemeProvider theme={theme}>
        <Drawer open={true} onClose={jest.fn} permanent={false}/>
      </ThemeProvider>,
  );

  expect(baseElement).toMatchSnapshot();
});

it('renders and matches snapshot with open and ' +
    'permanent without crashing', () => {
  const {container} = render(
      <ThemeProvider theme={theme}>
        <Drawer open={true} onClose={jest.fn} permanent={true}/>
      </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});

it('renders and matches snapshot with not open ' +
    'and permanent without crashing', () => {
  const {container} = render(
      <ThemeProvider theme={theme}>
        <Drawer open={false} onClose={jest.fn} permanent={true}/>
      </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});

it('renders and matches snapshot with not open and ' +
    'not permanent without crashing', () => {
  const {baseElement} = render(
      <ThemeProvider theme={theme}>
        <Drawer open={false} onClose={jest.fn} permanent={false}/>
      </ThemeProvider>,
  );

  expect(baseElement).toMatchSnapshot();
});
