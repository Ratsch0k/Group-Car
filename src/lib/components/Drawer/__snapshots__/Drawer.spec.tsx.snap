// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if user is logged in renders create group button if user has no groups 1`] = `
<body
  style=""
>
  <div>
    <div
      class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-69 makeStyles-drawerPermanent-71"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-70 makeStyles-drawerPaperPermanent-72 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      >
        <div
          class="MuiBox-root MuiBox-root-77 makeStyles-container-75"
        >
          <div
            class="MuiContainer-root makeStyles-body-74 MuiContainer-maxWidthLg"
          >
            <div
              class="MuiBox-root MuiBox-root-81 makeStyles-root-78"
            >
              <div
                class="MuiBox-root MuiBox-root-82 makeStyles-btn-79"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    drawer.createGroup
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiBox-root MuiBox-root-83 makeStyles-cars-80"
              >
                <div
                  class="MuiGrid-root makeStyles-root-84 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-85 makeStyles-footer-73"
          >
            <div
              class="MuiPaper-root makeStyles-footer-73 makeStyles-paper-86 MuiPaper-elevation6"
            >
              <div
                class="MuiContainer-root makeStyles-container-87 MuiContainer-maxWidthLg"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                    >
                      privacyPolicy.title
                    </button>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                    >
                      imprint.title
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`if user is logged in renders list of cars correctly if a group is selected 1`] = `
<body
  style=""
>
  <div>
    <div
      class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-112 makeStyles-drawerPermanent-114"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-113 makeStyles-drawerPaperPermanent-115 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      >
        <div
          class="MuiBox-root MuiBox-root-120 makeStyles-container-118"
        >
          <div
            class="MuiContainer-root makeStyles-body-117 MuiContainer-maxWidthLg"
          >
            <div
              class="MuiBox-root MuiBox-root-124 makeStyles-root-121"
            >
              <div
                class="MuiBox-root MuiBox-root-125 makeStyles-btn-122"
              >
                <div
                  class="MuiBox-root MuiBox-root-128"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Group
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-129 makeStyles-cars-123"
              >
                <div
                  class="MuiGrid-root makeStyles-root-130 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
                    >
                      <div
                        class="MuiCardHeader-root"
                      >
                        <div
                          class="MuiCardHeader-avatar"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root makeStyles-avatarIcon-138 MuiSvgIcon-fontSizeLarge"
                            color="Red"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiCardHeader-content"
                        >
                          <span
                            class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                          >
                            <b>
                              Car1
                            </b>
                          </span>
                          <span
                            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                          >
                            <div
                              class="MuiChip-root MuiChip-colorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-140"
                            >
                              <span
                                class="MuiChip-label MuiChip-labelSmall"
                              >
                                misc.available
                              </span>
                            </div>
                          </span>
                        </div>
                      </div>
                      <div
                        class="MuiCardActions-root makeStyles-actionsRoot-131 makeStyles-actionsContainer-132 MuiCardActions-spacing"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
                        >
                          <div
                            class="MuiGrid-root makeStyles-leftAction-134 MuiGrid-item MuiGrid-grid-xs-6"
                          >
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-fullWidth"
                              tabindex="0"
                              type="button"
                            >
                              <span
                                class="MuiButton-label"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M18.92 5.01C18.72 4.42 18.16 4 17.5 4h-11c-.66 0-1.21.42-1.42 1.01L3 11v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 15c-.83 0-1.5-.67-1.5-1.5S5.67 12 6.5 12s1.5.67 1.5 1.5S7.33 15 6.5 15zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 10l1.5-4.5h11L19 10H5z"
                                  />
                                </svg>
                                drawer.cars.drive
                              </span>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                          >
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled MuiButton-fullWidth Mui-disabled"
                              disabled=""
                              tabindex="-1"
                              type="button"
                            >
                              <span
                                class="MuiButton-label"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                                  />
                                </svg>
                                drawer.cars.view
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
                    >
                      <div
                        class="MuiCardHeader-root"
                      >
                        <div
                          class="MuiCardHeader-avatar"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root makeStyles-avatarIcon-138 MuiSvgIcon-fontSizeLarge"
                            color="grey"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiCardHeader-content"
                        >
                          <span
                            class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
                          >
                            <b>
                              Car2
                            </b>
                          </span>
                          <h6
                            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                          >
                            drawer.cars.drivenBy
                          </h6>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-141 makeStyles-footer-116"
          >
            <div
              class="MuiPaper-root makeStyles-footer-116 makeStyles-paper-142 MuiPaper-elevation6"
            >
              <div
                class="MuiContainer-root makeStyles-container-143 MuiContainer-maxWidthLg"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                    >
                      privacyPolicy.title
                    </button>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                    >
                      imprint.title
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`if user is logged in renders select group button if user has at least one group and none is selected 1`] = `
<body
  style=""
>
  <div>
    <div
      class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-88 makeStyles-drawerPermanent-90"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-89 makeStyles-drawerPaperPermanent-91 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      >
        <div
          class="MuiBox-root MuiBox-root-96 makeStyles-container-94"
        >
          <div
            class="MuiContainer-root makeStyles-body-93 MuiContainer-maxWidthLg"
          >
            <div
              class="MuiBox-root MuiBox-root-100 makeStyles-root-97"
            >
              <div
                class="MuiBox-root MuiBox-root-101 makeStyles-btn-98"
              >
                <div
                  class="MuiBox-root MuiBox-root-104"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      drawer.groupMenu.noSelection
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-105 makeStyles-cars-99"
              >
                <div
                  class="MuiGrid-root makeStyles-root-106 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-107 makeStyles-footer-92"
          >
            <div
              class="MuiPaper-root makeStyles-footer-92 makeStyles-paper-108 MuiPaper-elevation6"
            >
              <div
                class="MuiContainer-root makeStyles-container-109 MuiContainer-maxWidthLg"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                    >
                      privacyPolicy.title
                    </button>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                    >
                      imprint.title
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders NotLoggedIn component if user is not logged in 1`] = `
<body
  style=""
>
  <div />
</body>
`;

exports[`renders and matches snapshot with not open and not permanent without crashing 1`] = `
<body
  style=""
>
  <div />
</body>
`;

exports[`renders and matches snapshot with not open and permanent without crashing 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-37 makeStyles-drawerPermanent-39"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-38 makeStyles-drawerPaperPermanent-40 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
    >
      <div
        class="MuiBox-root MuiBox-root-45 makeStyles-container-43"
      >
        <div
          class="MuiBox-root MuiBox-root-48 makeStyles-fullHeight-47 makeStyles-center-46"
        >
          <div
            class="MuiBox-root MuiBox-root-49 makeStyles-center-46"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
            >
              drawer.notLoggedIn.title
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            >
              drawer.notLoggedIn.message
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-50 makeStyles-footer-41"
        >
          <div
            class="MuiPaper-root makeStyles-footer-41 makeStyles-paper-51 MuiPaper-elevation6"
          >
            <div
              class="MuiContainer-root makeStyles-container-52 MuiContainer-maxWidthLg"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                  >
                    privacyPolicy.title
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                  >
                    imprint.title
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders and matches snapshot with open and not permanent without crashing 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDrawer-root MuiDrawer-modal makeStyles-drawer-1"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorRight MuiPaper-elevation16"
      tabindex="-1"
    >
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-container-7"
      >
        <div
          class="MuiBox-root MuiBox-root-10 makeStyles-header-8"
        >
          <div
            class="MuiBox-root MuiBox-root-13 makeStyles-root-12"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root makeStyles-closeButton-11"
              data-testid="close"
              id="drawer-close"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-16 makeStyles-fullHeight-15 makeStyles-center-14"
        >
          <div
            class="MuiBox-root MuiBox-root-17 makeStyles-center-14"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
            >
              drawer.notLoggedIn.title
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            >
              drawer.notLoggedIn.message
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-18 makeStyles-footer-5"
        >
          <div
            class="MuiPaper-root makeStyles-footer-5 makeStyles-paper-19 MuiPaper-elevation6"
          >
            <div
              class="MuiContainer-root makeStyles-container-20 MuiContainer-maxWidthLg"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                  >
                    privacyPolicy.title
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                  >
                    imprint.title
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`renders and matches snapshot with open and permanent without crashing 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-21 makeStyles-drawerPermanent-23"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-22 makeStyles-drawerPaperPermanent-24 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
    >
      <div
        class="MuiBox-root MuiBox-root-29 makeStyles-container-27"
      >
        <div
          class="MuiBox-root MuiBox-root-32 makeStyles-fullHeight-31 makeStyles-center-30"
        >
          <div
            class="MuiBox-root MuiBox-root-33 makeStyles-center-30"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
            >
              drawer.notLoggedIn.title
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            >
              drawer.notLoggedIn.message
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-34 makeStyles-footer-25"
        >
          <div
            class="MuiPaper-root makeStyles-footer-25 makeStyles-paper-35 MuiPaper-elevation6"
          >
            <div
              class="MuiContainer-root makeStyles-container-36 MuiContainer-maxWidthLg"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                  >
                    privacyPolicy.title
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
                  >
                    imprint.title
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
